app/src/main/java/org/eyeseetea/malariacare/domain/boundary/io/IFileDownloader.java

```diff
@@ -1,18 +1,25 @@
 package org.eyeseetea.malariacare.domain.boundary.io;
 
+import org.eyeseetea.malariacare.domain.entity.Media;
 import org.eyeseetea.malariacare.domain.exception.FileDownloadException;
 
 import java.util.HashMap;
 import java.util.List;
 
 public interface IFileDownloader {
-
     interface Callback {
-        void onSuccess(HashMap<String, String>syncedFiles);
+        void onSuccess(HashMap<String, String> syncedFiles);
 
         void onError(FileDownloadException throwable);
 
-        void onRemove(String uid);
+        void removeRemotelyDeletedMedia(List<String> uids);
```

Comment:
the unique responsibility of FileDownloader should be download media, then is not its responsibility the save o delete media in database decision, this is a use case decision. For use case can decide to save or delete then onSuccess callback should return media and not hashmap. 
Then we should remove removeRemotelyDeletedMedia and save methods from IFileDownloader callbacks